python/ray/train/v2/_internal/state/state_actor.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import threading
+from collections import defaultdict
+from typing import Dict, Optional
+
+import ray
+from ray.actor import ActorHandle
+from ray.train.v2._internal.state.schema import TrainRun, TrainRunAttempt
+
+logger = logging.getLogger(__name__)
+
+
+class TrainStateActor:
+    def __init__(self):
+        self._runs: Dict[str, TrainRun] = {}
+        # {run_id: {attempt_id: TrainRunAttempt}}
+        self._run_attempts: Dict[str, Dict[str, TrainRunAttempt]] = defaultdict(dict)
+
+    def create_or_update_train_run(self, run: TrainRun) -> None:
+        self._runs[run.id] = run
+
+    def create_or_update_train_run_attempt(self, run_attempt: TrainRunAttempt):
+        self._run_attempts[run_attempt.run_id][run_attempt.attempt_id] = run_attempt
+
+    def get_train_runs(self) -> Dict[str, TrainRun]:
+        return self._runs
+
+    def get_train_run_attempts(self) -> Dict[str, Dict[str, TrainRunAttempt]]:
+        return self._run_attempts
+
+
+class PrintTrainStateActor(TrainStateActor):
+    def create_or_update_train_run(self, run: TrainRun) -> None:
+        super().create_or_update_train_run(run)
+        print(f"[RUN] {run.id=} {run.status.name=} {run.status_detail=}")
+
+    def create_or_update_train_run_attempt(self, run_attempt: TrainRunAttempt):
+        super().create_or_update_train_run_attempt(run_attempt)
+        print(
+            f"[ATTEMPT] {run_attempt.run_id=} {run_attempt.attempt_id=} {run_attempt.status.name=} {run_attempt.status_detail=}"
+        )
+
+    def get_train_runs(self) -> Dict[str, TrainRun]:
+        return super().get_train_runs()
+
+    def get_train_run_attempts(self) -> Dict[str, Dict[str, TrainRunAttempt]]:
+        return super().get_train_run_attempts()
+
+
+TRAIN_STATE_ACTOR_NAME = "train_v2_state_actor"
+TRAIN_STATE_ACTOR_NAMESPACE = "_train_state_actor"
+
+_state_actor_lock: threading.RLock = threading.RLock()
+
+
+def get_or_create_state_actor() -> ActorHandle:
+    """Get or create the Ray Train State Actor."""
+
+    with _state_actor_lock:
+        state_actor = (
+            ray.remote(PrintTrainStateActor)
+            .options(
+                num_cpus=0,
+                name=TRAIN_STATE_ACTOR_NAME,
+                namespace=TRAIN_STATE_ACTOR_NAMESPACE,
+                get_if_exists=True,
+                lifetime="detached",
+                resources={"node:__internal_head__": 0.001},
+                # Escape from the parent's placement group
+                scheduling_strategy="DEFAULT",
```

Comment:
Should we enable unlimited actor restarts? `ray.remote(num_cpus=0, max_restarts=-1, max_task_retries=-1)`